File: FragmentSample/app/src/main/java/com/example/app/MyApplication.java
/**
 * Copyright 2021. Happy codding ! :)
 * Author: Serhii Butryk
 */
package com.example.app;

import android.app.Application;
import android.util.Log;

import static com.example.app.utils.AppUtils.APP;

/**
 * Custom app's application class
 */
public class MyApplication extends Application {

    public static final String TAG = APP + "MyApplication";

    /**
     * no-op, logging only
     */
    @Override
    public void onCreate() {
        super.onCreate();
        Log.i(TAG, "onCreate() Application is created");
    }
}

File: ServiceSample/app/src/main/java/com/example/servicesample/MyApplication.kt
/**
 * Copyright 2022. Happy codding ! :)
 * Author: Serhii Butryk
 */
package com.example.servicesample

import android.app.Application
import android.util.Log

/**
 * App's application class. No-op, only prints logs.
 */
class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        Log.i(TAG, "onCreate: application is created - $this")
    }

    companion object {
        /**
         * Tag for filtering logs of this sample app
         */
        const val APP_TAG = "ServiceSample"
        private const val TAG = "$APP_TAG-MyApplication"
    }

}